hydroflow_lang/src/graph/ops/defer_tick.rs

```diff
@@ -3,26 +3,45 @@ use super::{
     IDENTITY_WRITE_FN, RANGE_0, RANGE_1,
 };
 
-/// Delays all elements which pass through to the next tick. In short,
-/// execution of a hydroflow graph runs as a sequence of distinct "ticks".
-/// Non-monotonic operators compute their output in terms of each tick so
-/// execution doesn't have to block, and it is up to the user to coordinate
-/// data between tick executions to achieve the desired result.
-///
-/// An tick may be divided into multiple _strata_, see the [`next_stratum()`](#next_stratum)
+/// Buffers all input items and releases them in the next tick.
+/// the state of the current tick. For example,
+/// See the [book discussion of Hydroflow time](../concepts/life_and_times) for details on ticks.
+/// A tick may be divided into multiple [strata](../concepts/stratification); see the [`next_stratum()`](#next_stratum)
 /// operator.
 ///
+/// `defer_tick` is sometimes needed to separate conflicting data across time,
+/// in order to preserve invariants. Consider the following example, which implements
+/// a flip-flop -- the invariant is that it emit one of true or false in a given tick
+/// (but never both!)
+/// pub fn main() {
```

Comment:
I think you want to wrap this example in ```rustbook

Reply:
You could also write this example using ```hydroflow and use -> assert(|x| if context.current_tick() % 2 { x == true } else { x == false }) instead of having all the rust code in there.

Reply:
nice!